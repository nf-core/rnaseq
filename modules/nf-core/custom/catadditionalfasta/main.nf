process CUSTOM_CATADDITIONALFASTA {
    tag "$meta.id"

    conda "conda-forge::python=3.9.5"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
        'biocontainers/python:3.9--1' }"

    input:
    tuple val(meta), path(fasta), path(gtf)
    tuple val(meta2), path(add_fasta)
    val  biotype

    output:
    tuple val(meta), path("*/*.fasta") , emit: fasta
    tuple val(meta), path("*/*.gtf")   , emit: gtf
    path "versions.yml"                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'fasta2gtf.py'
}
