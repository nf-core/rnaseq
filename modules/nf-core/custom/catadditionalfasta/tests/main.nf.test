nextflow_process {

    name "Test Process CUSTOM_CATADDITIONALFASTA"
    script "../main.nf"
    process "CUSTOM_CATADDITIONALFASTA"

    tag "modules"
    tag "modules_nfcore"
    tag "custom"
    tag "custom/catadditionalfasta"

    test("sarscov2 - fastq - gtf") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], 
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true),
                    file(params.test_data['sarscov2']['genome']['genome_gtf'], checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test', single_end:false ], 
                    file(params.test_data['sarscov2']['genome']['transcriptome_fasta'], checkIfExists: true)
                ]
                input[2] = 'test_biotype'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.fasta).match("fasta") },
                { assert snapshot(process.out.gtf).match("gtf") },
                { assert snapshot(process.out.versions).match("versions") }
            )
        }

    }

}
