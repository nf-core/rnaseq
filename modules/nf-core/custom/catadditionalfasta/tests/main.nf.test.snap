{
    "versions": {
        "content": [
            [
                "versions.yml:md5,26f47339777a265af57338ac7f0f8798"
            ]
        ],
        "timestamp": "2023-12-20T20:52:55.242485"
    },
    "gtf": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "genome_transcriptome.gtf:md5,bc88d95e7f27540e6b9906105d5be361"
                ]
            ]
        ],
        "timestamp": "2023-12-20T19:46:31.839377"
    },
    "fasta": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "genome_transcriptome.fasta:md5,6a20c1a2e465519320a0d01f338f5cb5"
                ]
            ]
        ],
        "timestamp": "2023-12-20T19:42:47.12194"
    }
}